blueocean-config/src/main/java/io/jenkins/blueocean/config/JenkinsJSExtensions.java

```diff
@@ -122,54 +109,86 @@ private synchronized static void refreshCache(List<PluginWrapper> latestPlugins)
             refreshCache(pluginCache);
         }
         for (PluginWrapper pluginWrapper : pluginCache) {
+            //skip if not active
+            if (!pluginWrapper.isActive()) {
+                continue;
+            }
             //skip probing plugin if already read
             if (jsExtensionCache.get(pluginWrapper.getLongName()) != null) {
                 continue;
             }
             try {
                 Enumeration<URL> dataResources = pluginWrapper.classLoader.getResources("jenkins-js-extension.json");
+                boolean hasDefinedExtensions = false;
+
                 while (dataResources.hasMoreElements()) {
                     URL dataRes = dataResources.nextElement();
-                    StringWriter fileContentBuffer = new StringWriter();
 
                     LOGGER.debug("Reading 'jenkins-js-extension.json' from '{}'.", dataRes);
 
-                    try {
-                        IOUtils.copy(dataRes.openStream(), fileContentBuffer, Charset.forName("UTF-8"));
-                        Map<?,List<Map>> extensionData = mapper.readValue(dataRes.openStream(), Map.class);
-                        List<Map> extensions = (List<Map>)extensionData.get("extensions");
+                    try (InputStream dataResStream = dataRes.openStream()) {
+                        Map<String, Object> extensionData = mapper.readValue(dataResStream, Map.class);
+
+                        String pluginId = getGav(extensionData);
+                        if (pluginId != null) {
+                            // Skip if the plugin name specified on the extension data does not match the name
+                            // on the PluginWrapper for this iteration. This can happen for e.g. aggregator
+                            // plugins, in which case you'll be seeing extension resources on it's dependencies.
+                            // We can skip these here because we will process those plugins themselves in a
+                            // future/past iteration of this loop.
+                            if (!pluginId.equals(pluginWrapper.getShortName())) {
```

Comment:
Not sure. There was a bit of a rewrite done on this at some stage ... it introduced a number of bugs + made it a lot less efficient. That said ... not sure if this was ever done.